gpii/node_modules/settingsHandlers/src/WebSocketsComponent.js

```diff
@@ -0,0 +1,262 @@
+
+        var results = [];
+
+        for (var i=0; i< payload.length; i++) {
+            /*jshint -W083 */
+            results[i] = {settings: {}};
+            results[i].settings = fluid.transform(payload[i].settings, function (value, key) {
+                var currentValue = fluid.get(currentSettings, key);
+                return currentValue;
+            });
+        }
+
+        return results;
+    };
+
+    gpii.settingsHandlers.WebSocketsComponent.get = function (payload, that) {
+        var result = fluid.copy(payload);
+
+        for (var appId in payload) {
+            if (appId in that.model.settings) {
+                result[appId] = gpii.settingsHandlers.WebSocketsComponent.getSettings(payload[appId], appId, that);
+            //} else {
+            //    result[appId] = {settings: {}};
+            }
+        }
+        return result;
+    };
+
+    gpii.settingsHandlers.WebSocketsComponent.setSettingsIndividually = function (payload, solutionId, that) {
+       var results = {settings: {}};
+       var solution = solutionId.replace(/\./g, "\\.");
+
+       // How can I set these options to the applier in a declarative way?
+       //
+       that.applier.options.resolverGetConfig = fluid.model.escapedGetConfig;
+       that.applier.options.resolverSetConfig = fluid.model.escapedSetConfig;
+
+       fluid.each(payload.settings, function (value, key) {
+           // 1st part of the work around the problems we're having when setting back some settings
+           // to undefined, ie: a key called "like.this"
+           //
+           var solutionIdAndKey = [solutionId];
+           var keyWithDots = false;
+
+           if (key.match("\\.")) {
+               keyWithDots = true;
+               var s = key.split(".");
+               solutionIdAndKey = solutionIdAndKey.concat(s);
+           } else {
+               solutionIdAndKey = solutionIdAndKey.concat(key);
+           }
+
+           var currentValue = fluid.get(that.model.settings, solutionIdAndKey);
+
+           var type;
+           if (currentValue === undefined) {
+               type = "ADD";
+           } else if (value === undefined) {
+               type = "DELETE";
+           } else {
+               type = null;
+           }
+
+           results.settings[key] = {
+               oldValue: currentValue,
+               newValue: value,
+           };
+
+           that.applier.change("settings." + solution + "." + key, value, type);
+
+           //2nd part of the workaround
+           //
+           if (keyWithDots && type === "DELETE") {
+               var s = key.split(".");
+               for (var i = s.length; i > 0; i--) {
+                   if (i === s.length) {continue};
+                   var v = fluid.get(that.model.settings, solutionIdAndKey.slice(0, i + 1));
+                   if (Object.keys(v).length === 0) {
+                       var objToRemove = s.slice(0, i).toString().replace(/\,/g, ".");
```

Comment:
This horrible workaround is a consequence of the problems that I'm experiencing with https://github.com/GPII/universal/pull/303#discussion_r32400750